### DISCUSSION TITLE
Options pay off diagram

### DISCUSSION THREAD

**Discussion by pannet1** [highfestiva/finplot#437] [Ideas]
Is there a plan to add support for options payoff diagram

**Comment by highfestiva** [comment]
All such things can currently be done, but you have to draw the lines yourself depending on the contract, commission, etc. finplot only forms the basis for financial charts, whereas indicators, profit & loss, courtage vary between assets, markets, exchanges, rebates, etc.

That said, it should be very easy for you to implement yourself.

**Reply by pannet1** [reply]
wow. i was under the impression that it is a special kind of diagram. can you include an example if possible.

**Reply by highfestiva** [reply]
Here are some, as you see straight lines, possibly with a curved one for straddle: https://www.optiontradingtips.com/options101/payoff-diagrams.html

**Reply by pannet1** [reply]
nice to see that it is already used in production. however, what i meant to ask is an example [here](https://github.com/highfestiva/finplot/tree/master/finplot/examples)

**Reply by highfestiva** [reply]
No, the intention of the examples is to show _how_ to draw, not _what_ to draw, as there are infinite possibilities depending the trader's style, assets, contracts, etc. Try learning a tiny bit of Python, then going through the examples, make your own changes and you'll have the basics required figured out in no time.

Try this to get started. Resembles what's drawn in the page I linked.

```python
#!/usr/bin/env python3

import finplot as fplt
import pandas as pd
import numpy as np


rng = lambda a, b: pd.Series(np.linspace(a, b, 40) if a!=b else [a]*40)

x = pd.date_range('01:00', '20:30', periods=80)
buy_put  = pd.concat([rng(400, -100), rng(-100, -100)]).reset_index(drop=True)
buy_call = pd.concat([rng(-100, -100), rng(-100, 400)]).reset_index(drop=True)
payoff = pd.concat([rng(300, -200), rng(-200, 300)]).reset_index(drop=True)
pnl = payoff.rolling(31, center=True).mean()

fplt.create_plot('Some Payoff Chart')

fplt.plot(x, buy_put, style='- ', width=2, legend='Buy Put Option')
fplt.plot(x, buy_call, style='- ', width=2, legend='Buy Call Option')
fplt.plot(x, payoff, width=2, legend='Straddle Payoff')
fplt.plot(x, pnl, width=2, legend='Straddle P&L')

fplt.show()
```

![image](https://github.com/highfestiva/finplot/assets/160303/393b7fdf-70d6-4fa4-949d-e1616b68d555)

**Reply by pannet1** [reply]
coding full time for indian stock market. there is no requirement for me now, but sure plotting options payoff diagram is a good skillset to have. i asked because there is rage for options trading in our market and will get people  excited about finplot. thank you @highfestiva.
